src/P256.sol

```diff
@@ -15,10 +16,16 @@ library P256 {
         uint256 y
     ) internal view returns (bool) {
         bytes memory args = abi.encode(message_hash, r, s, x, y);
-        (bool success, bytes memory ret) = VERIFIER.staticcall(args);
-        assert(success); // never reverts, always returns 0 or 1
 
-        return abi.decode(ret, (uint256)) == 1;
+        (bool success, bytes memory ret) = PRECOMPILE.staticcall(args);
+        if (success && ret.length > 0) return abi.decode(ret, (uint256)) == 1;
```

Comment:
This check will pass if the precompile is not deployed on a given chain...

Just confirmed that an undeployed contract (codesize 0) will return success=true, ret=empty array. And codesize will return 0 for the precompile address whether you're on a 7212 chain or not.

I think RIP-7212 is flawed and needs to be fixed